assignment/assignment.hs

```diff
@@ -45,6 +52,30 @@ import Data.Char(intToDigit)
 
 -- myor:: [ Bool ] -> Bool 
 
+--andOr func first second = 
+--  if first func second
+--    then True
+--    else False
+
+--myand list = foldl (allTrue (&&)) True list
+--myor list = foldl (andOr (||)) False list
+
+--not really sure what went wrong here so I just have to do it a longer way
+
+allTrue first second = 
+  if first && second
+    then True
+    else False
+
+myand list = foldl (allTrue) True list
+
+anyTrue first second = 
+  if first || second
+    then True
+    else False
+
+myor list = foldl (anyTrue) False list
+
 
```

Comment:
So this is a pet  peeve.

This about 

``` hs
  if first && second
    then True
    else False
```

The condition `first && second` returns `True` if they are both true. So there's no need to test with an if: you could just write

~~~ hs
allTrue first second = 
  first && second
~~~

But this is a two parameter function that calls another two parameter function, so it's the same as

~~~ hs
allTrue  = 
  (&&)
~~~

Which means we could rewrite your fold as:

~~~ hs
myand = foldl (&&) True
~~~

:)